packages/react/src/template.tsx

```diff
@@ -32,16 +32,20 @@ export type AsyncHandlerEventDataState = {
   $: 'event'
   key: string
   async: true
+  // server-only
+  handler?: EventHandler
 }
 export type HandlerEventDataState = {
   $: 'event'
   key: string
   async: false
+  // server-only
+  handler?: EventHandler
```

Comment:
hmm.. not super happy about this, but I'm glad it works

we could alternatively have a higher level server state that has `EventHandlerState & { handler: ...}`

Reply:
yeah, this is types thing. I will have to rework it, but otherwise I think it's a good way.